compiler/rustc_ast_lowering/src/path.rs

```diff
@@ -226,11 +226,10 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
 
     pub(crate) fn lower_use_path(
         &mut self,
-        res: SmallVec<[Res; 3]>,
+        res: PerNS<Option<Res>>,
         p: &Path,
         param_mode: ParamMode,
     ) -> &'hir hir::UsePath<'hir> {
-        assert!((1..=3).contains(&res.len()));
```

Comment:
The assert checked that the `res` is not empty, in particular (that means HIR visitor will visit the path at least once), better to keep it.

Reply:
Good point, though we can use `!res.is_empty()` now.